tests/serverpod_test_server/test_e2e/method_streaming/out_stream_test.dart

```diff
@@ -76,4 +76,38 @@ void main() {
       isA<ExceptionWithData>(),
     );
   });
+
+  test(
+      'Given a streaming method that is not a generator that throws an exception before the stream is returned, then stream is closed with ServerpodClientException.',
+      () async {
+    var streamComplete = Completer<dynamic>();
+    var stream = client.methodStreaming.exceptionThrownBeforeStreamReturn();
+    stream.listen((event) {
+      // Do nothing
+    }, onError: (error) {
+      streamComplete.complete(error);
+    });
+
+    await expectLater(
+      await streamComplete.future,
+      isA<ConnectionClosedException>(),
+    );
```

Comment:
Question: Do you think it's possible to omit the completer and use built-in assertion for this? Same for next test.
```dart
await expectLater(stream, emitsError(isA<ConnectionClosedException>))
```

Reply:
Updated all the tests in the file to this pattern.